google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/sql/Statement.java

```diff
@@ -191,28 +242,120 @@ private static Value booleanParamOf(@Nullable Boolean value) {
     private static Value timestampParamOf(@Nullable Instant value) {
       Value.Builder builder = nullValueWithType(TIMESTAMP_TYPE);
       if (value != null) {
-        builder.setTimestampValue(
-            Timestamp.newBuilder()
-                .setSeconds(value.getEpochSecond())
-                .setNanos(value.getNano())
-                .build());
+        builder.setTimestampValue(toTimestamp(value));
       }
       return builder.build();
     }
 
     private static Value dateParamOf(@Nullable Date value) {
       Value.Builder builder = nullValueWithType(DATE_TYPE);
       if (value != null) {
-        builder.setDateValue(
-            com.google.type.Date.newBuilder()
-                .setYear(value.getYear())
-                .setMonth(value.getMonth())
-                .setDay(value.getDayOfMonth())
-                .build());
+        builder.setDateValue(toProtoDate(value));
+      }
+      return builder.build();
+    }
+
+    private static <T> Value arrayParamOf(@Nullable List<T> value, SqlType.Array<T> arrayType) {
+      Type type =
+          Type.newBuilder()
+              .setArrayType(
+                  Type.Array.newBuilder().setElementType(getElementType(arrayType)).build())
+              .build();
+      Value.Builder builder = nullValueWithType(type);
+      if (value != null) {
+        builder.setArrayValue(arrayValueOf(value, arrayType));
       }
       return builder.build();
     }
 
+    private static Type getElementType(SqlType.Array<?> arrayType) {
+      switch (arrayType.getElementType().getCode()) {
+        case BYTES:
+          return BYTES_TYPE;
+        case STRING:
+          return STRING_TYPE;
+        case INT64:
+          return INT64_TYPE;
+        case FLOAT32:
+          return FLOAT32_TYPE;
+        case FLOAT64:
+          return FLOAT64_TYPE;
+        case BOOL:
+          return BOOL_TYPE;
+        case TIMESTAMP:
+          return TIMESTAMP_TYPE;
+        case DATE:
+          return DATE_TYPE;
+        default:
+          throw new IllegalArgumentException(
+              "Unsupported query parameter Array element type: " + arrayType.getElementType());
+      }
+    }
+
+    private static ArrayValue arrayValueOf(List<?> value, SqlType.Array<?> arrayType) {
+      ArrayValue.Builder valueBuilder = ArrayValue.newBuilder();
+      for (Object element : value) {
+        if (element == null) {
```

Comment:
Should this be of a `nullValueWithType`?

Reply:
No, the API only wants the type set on the outermost value, not nested elements